lookout/style/format/benchmarks/compare_quality_reports.py

```diff
@@ -0,0 +1,81 @@
+from typing import Union
+
+import pandas
+from tabulate import tabulate
+
+from lookout.style.format.benchmarks.top_repos_quality import FLOAT_PRECISION
+
+
+def _convert_cell_to_value(cell: str) -> Union[int, float, str]:
+    for type_ in (int, float):
+        try:
+            return type_(cell.strip())
+        except ValueError:
+            pass
+    cell = cell.strip()
+    return cell if cell else float("nan")
+
+
+def _quality_report_table_to_df(report_table_path: str) -> pandas.DataFrame:
+    table = []
+    separator_set = frozenset(":|-")
+    with open(report_table_path) as f:
+        for line in f:
+            if set(line.strip()) <= separator_set:
+                continue
+            cells = line.strip().split("|")
+            if not cells[0]:
+                cells = cells[1:]
+            if not cells[-1]:
+                cells = cells[:-1]
+            values = [_convert_cell_to_value(c) for c in cells]
+            table.append(values)
+    return pandas.DataFrame(table[1:], columns=table[0])
+
+
+def compare_quality_reports(base: str, new: str, output: str) -> None:
```

Comment:
All the three should be file objects; you can further add `compare_quality_reports_with_paths` which accepts strings and opens all the files.

Reply:
ok. now I see what you want.